contracts/aave/AaveV2Contract.sol

```diff
@@ -0,0 +1,173 @@
+     */
+    function getLatestPrice(address _asset) public view returns (int256 price) {
+        AggregatorV3Interface priceFeed = AggregatorV3Interface(_asset);
+        (, price, , , ) = priceFeed.latestRoundData();
+    }
+
+    function nomalize(uint256 amount) public view returns (uint256) {
+        return amount * 10**(ETH_DECIMALS - USD_DECIMALS);
+    }
+
+    //----------------------------//
+    //     Mutation Functions     //
+    //----------------------------//
+
+    /**
+     * @dev Only owner can add new assets that the contract will support for staking
+     * @param _asset will be address of the asset
+     * @param _priceOracle will the chainlink asset price oracle
+     * @return event AddAsset(_asset: 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2, _priceOracle: 0xAc559F25B1619171CbC396a50854A3240b6A4e99)
+     */
+    function addDepositAsset(
+        address _asset,
+        address _priceOracle
+    ) public onlyOwner isValid( _asset) {
+        depositAssets[_asset] = _priceOracle;
+        emit AddAsset(_asset,_priceOracle);
+    }
+
+    /**
+     * @dev Users will deposit their assets in pool
+     * @param _asset will be asset name like ETH or BTC
+     * @param _amount will be the deposited amount
+     */
+    function deposit( address _asset,uint256 _amount) public nonReentrant payable {
+        require(
+            depositAssets[_asset] != address(0),
+            Errors.ASSET_NOT_ALLOWED
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+
+        if (msg.value > 0) {
+            WETHGATEWAY.depositETH{value: msg.value}(
+                address(POOL),
+                address(this),
+                0
+            );
+        } else {
+            require(_amount > 0, Errors.ZERO_AMOUNT);
+            IERC20 supplyAsset = IERC20(_asset);
+            supplyAsset.transferFrom(msg.sender, address(this), _amount);
+            supplyAsset.approve(address(POOL), _amount);
+            POOL.deposit(address(supplyAsset), _amount, address(this), 0);
+        }
+    }
+
+    /**
+     * @dev Cruize will borrow on behalf of user
+     * @param _asset will be asset name like ETH or BTC
+     */
+    function borrow(address _asset) public nonReentrant {
+        (, , uint256 availableBorrowsETH, , , ) = POOL.getUserAccountData(
+            address(this)
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+        uint256 availableBorrowAmountIn18Decimals = (nomalize(
+            uint256(assetPrice)
+        ) * availableBorrowsETH) / 1e18;
+        uint256 availableBorrowAmountIn6Decimals = availableBorrowAmountIn18Decimals /
+                1e12;
+        uint256 borrowAmount = (availableBorrowAmountIn6Decimals *
+            borrowRatio) / 1000;
+        POOL.borrow(address(BORROW_ASSET), borrowAmount, 2, 0, address(this));
+    }
+
+    /**
+     * @dev Cruize pool will repay the debt amount
+     */
+    function repay() public nonReentrant {
```

Comment:
there is no check condition if the contract has funds for pay or not .it must be here

Reply:
it's is also an public function mean anyone can pay

Reply:
no check for owner